### PR TITLE
Giraffe sign page social sharing

### LINKED ISSUE #327: Petition share links for redesign
On the thanks page, share links should be carried over from the legacy implementation.  Note that share links on that version are calculated partially from data gotten from the signature POST.

On the petition page itself, the new design includes four links:
 - [x] Facebook share - if there is a sharebandit url, upon clicking the link it should get `getSharebanditShareLink` the same way the thanks page does (ideally not before).  If there is no sharebandit link, it should be the standard facebook petition link, but with `source=n.fb.p` where `n` is for "non-signer" and `.p` is for the petition page.  Note that if the client 'knows' that the user is the creator or has signed the petition, then you can/should replace `n.` with either `c.` or `s.` appropriately
 - [x] Twitter share - This should use the lib `petitionShortCode` url, with mode=`n`.  Note that if the user is logged in or has guest credentials (e.g. akid, etc), then you can/should pass in the userId
 - [x] Email share - This will be the same thing as the thanks page except the link will be `source= n.mt.p` (unless it's meant to be copy-pasted, in which case n.cp.p)
 - [x] Copy Link share should be the same think as rawShareLink but with mode=`m` for a non-signer (keep the other two, if they are known creator/signers, but don't hit an additional api to find out).

See [this PR](https://github.com/MoveOnOrg/mop/pull/235) and the full shortcodes file for short-code related questions. 

Also update docs/EXPLANATION-share-links.md

### PR DESCRIPTION
**Pull request by jparkrr**
Depends on #330

Makes the social sharing links on the sign page work. See #327 for the explanation of the source prefixes and suffixes.

closes #327 
tracking issue #309

### CODE REVIEW

File: src/components/theme-giraffe/petition-share.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+))
+
+const Facebook = withFacebook(MoShare.Facebook)
+
+const Twitter = withTwitter(MoShare.Twitter)
+
+const CopyLink = withShareLink(({ rawLink, ...rest }) => (
+  <MoShare.CopyLink
+    {...rest}
+    onClick={() => prompt('Here is your share link:', rawLink)}
```
Comment (jparkrr) [inline src/components/theme-giraffe/petition-share.js]: I asked a question about this here: https://github.com/MoveOnOrg/giraffe/issues/79

**Comment by sreynen** [comment]
Marking this as blocked on the UX issue. Also has the same issue with short links I flagged in #333. Otherwise looks good to me.

**Comment by sreynen** [comment]
Looks like this is no longer blocked.

**Comment by jparkrr** [comment]
Added copy link here too. Should be good to go once https://github.com/MoveOnOrg/giraffe/pull/111 and https://github.com/MoveOnOrg/mop-frontend/pull/333 are merged

File: src/giraffe-ui/petition/share.js [hunk 2]
```diff
@@ -7,39 +7,17 @@ import TwitterSvg from '../svgs/twitter.svg'
...
-        </Link>
-        <Link onClick={tweet} className={elementCn}>
-          <TwitterSvg />
-          {hasLabels && 'Twitter'}
-        </Link>
-        <Link onClick={link} className={elementCn}>
-          <LinkSvg />
-          {hasLabels && 'Copy Link'}
-        </Link>
+      <div className={`${className}__share${hasLabels ? '__links' : '-icons'}`}>
```
Comment (schuyler1d) [inline src/giraffe-ui/petition/share.js]: this is another place where it might be useful to revisit the classname-choices as a library for the new theme.

**Pull request merged**